src/pdf.rs

```diff
@@ -0,0 +1,85 @@
+// qrxfil - exfiltrate files with QR codes
+// Copyright (C) 2021 Jb Doyon
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/>.
+
+//! Create PDFs for printed paper backup
+
+use printpdf::*;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::{Path, PathBuf};
+
+fn read_folder_sorted(folder: &Path) -> Vec<PathBuf> {
+    let mut output_files: Vec<PathBuf> = std::fs::read_dir(folder)
+        .expect("Could not list output directory")
+        .map(Result::unwrap)
+        .filter(|file| file.file_name().to_str().unwrap().ends_with("png"))
+        .map(|e| e.path())
+        .collect();
+    // read_dir does not guarantee ordering => explicit sort chunk files
+    output_files.sort();
+    output_files
+}
+
+pub fn gendoc(output_folder: &Path) {
+    let chunk_image_filenames = read_folder_sorted(output_folder);
+
+    let title = format!("QRXFIL export ({} chunks)", chunk_image_filenames.len());
+    let (doc, page1, layer1) = PdfDocument::new(title.clone(), Mm(21.0), Mm(29.7), "Layer 1");
+
+    let current_layer = doc.get_page(page1).get_layer(layer1);
+
+    let font = doc
+        .add_external_font(
+            File::open("/usr/share/fonts/truetype/firacode/FiraCode-Bold.ttf").unwrap(),
```

Comment:
Font I like from the `fonts-firacode` debian package. I don't care about the font itself that much, I just want "default text".

As it is, this code isn't runnable unless that precise package was installed (thinking of CI right now)